pasqal_cloud/batch.py

```diff
@@ -63,36 +68,38 @@ class Batch(BaseModel):
     user_id: str
     priority: int
     status: str
-    webhook: Optional[str]
-    _client: Client
+    _client: Client = PrivateAttr(default=None)
     sequence_builder: str
-    start_datetime: Optional[str]
-    end_datetime: Optional[str]
-    device_status: Optional[str]
     ordered_jobs: List[Job] = []
     jobs_count: int = 0
     jobs_count_per_status: Dict[str, int] = {}
+    webhook: Optional[str] = None
+    start_datetime: Optional[str] = None
+    end_datetime: Optional[str] = None
+    device_status: Optional[str] = None
     parent_id: Optional[str] = None
     configuration: Union[BaseConfig, Dict[str, Any], None] = None
 
-    class Config:
-        extra = Extra.allow
-        arbitrary_types_allowed = True
+    model_config = ConfigDict(extra="allow", arbitrary_types_allowed=True)
 
-    @root_validator(pre=True)
-    def _build_ordered_jobs(cls, values: Dict[str, Any]) -> Dict[str, Any]:
-        """This root validator will modify the 'jobs' attribute which is a list
+    def __init__(self, **data: Any) -> None:
+        """
+        Makes sure the '_client' is set when instantiating a Batch
+        as Pydantic V2 does not support private attributes.
+        """
+        super().__init__(**data)
+        self._client = data["_client"]
+
+    @model_validator(mode="before")
+    def _build_ordered_jobs(cls, data: Dict[str, Any]) -> Dict[str, Any]:
+        """This root validator will modify the 'jobs' attribute, which is a list
         of jobs dictionaries ordered by creation time before instantiation.
         It will duplicate the value of 'jobs' in a new attribute 'ordered_jobs'
         to keep the jobs ordered by creation time.
         """
-        ordered_jobs_list = []
-        jobs_received = values.get("jobs", [])
-        for job in jobs_received:
-            job_dict = {**job, "_client": values["_client"]}
-            ordered_jobs_list.append(job_dict)
-        values["ordered_jobs"] = ordered_jobs_list
-        return values
+        jobs_received = data.get("jobs", [])
+        data["ordered_jobs"] = [{**job, "_client": data["_client"]} for job in jobs_received]
```

Comment:
I feel like it will be too much in this case if I replace jobs_received, too many information on the same line. I think this format is clear and compact enough. Anyway I think this is more of a question of preference. If you think it is not relevant I can change it.

Reply:
The 2nd line at a glance is obscure so not sure readability or clarity means anything there. I'm just pointing out you can reduce it further if we want, you do what you prefer